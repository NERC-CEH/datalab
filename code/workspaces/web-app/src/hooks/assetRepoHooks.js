import useShallowSelector from './useShallowSelector';
import assetRepoSelectors from '../selectors/assetRepoSelectors';
import sortByName from '../components/common/sortByName';
import { PUBLIC } from '../components/assetRepo/assetVisibilities';

const useAssetRepoUnsorted = () => useShallowSelector(assetRepoSelectors.assetRepo);

export const useAssetRepo = () => {
  const assetRepo = useAssetRepoUnsorted();
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(assetRepo.value.assets),
    },
  };
};

const assetVisible = (projectKey, asset) => asset.fileLocation
  && (asset.visible === PUBLIC
  || (asset.projects && asset.projects.map(project => project.key).includes(projectKey)));

export const useVisibleAssets = (projectKey) => {
  const assetRepo = useAssetRepoUnsorted();
  const visibleAssets = assetRepo.value.assets
    ? assetRepo.value.assets.filter(asset => assetVisible(projectKey, asset))
    : [];
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(visibleAssets),
    },
  };
};
