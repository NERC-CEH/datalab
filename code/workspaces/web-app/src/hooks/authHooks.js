import useShallowSelector from './useShallowSelector';
import authSelectors from '../selectors/authSelectors';

const { currentUserId, currentUserPermissions, currentUserTokens } = authSelectors;

export const useCurrentUserId = () => useShallowSelector(currentUserId);
export const useCurrentUserPermissions = () => useShallowSelector(currentUserPermissions);
export const useCurrentUserTokens = () => useShallowSelector(currentUserTokens);
