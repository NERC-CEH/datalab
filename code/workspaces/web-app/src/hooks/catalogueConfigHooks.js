import useShallowSelector from './useShallowSelector';
import catalogueConfigSelectors from '../selectors/catalogueConfigSelectors';

export const useCatalogueConfig = () => useShallowSelector(catalogueConfigSelectors.catalogueConfig);
export const useCatalogueAvailable = () => useShallowSelector(catalogueConfigSelectors.catalogueAvailable);
