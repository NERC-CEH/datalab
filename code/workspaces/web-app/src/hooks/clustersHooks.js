import useShallowSelector from './useShallowSelector';
import clustersSelectors from '../selectors/clustersSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useClusters = () => useShallowSelector(clustersSelectors.clusters);
export const useClustersByType = (type) => {
  const clusters = useClusters();

  if (!type) {
    return clusters;
  }

  return {
    ...clusters,
    value: clusters.value.filter(cluster => cluster.type === type),
  };
};
