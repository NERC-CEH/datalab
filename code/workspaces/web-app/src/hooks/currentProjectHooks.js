import useShallowSelector from './useShallowSelector';
import currentProjectSelectors from '../selectors/currentProjectSelectors';

const { currentProject, currentProjectKey } = currentProjectSelectors;

export const useCurrentProject = () => useShallowSelector(currentProject);
export const useCurrentProjectKey = () => useShallowSelector(currentProjectKey);
