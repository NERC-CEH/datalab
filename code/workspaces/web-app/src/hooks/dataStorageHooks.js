import useShallowSelector from './useShallowSelector';
import dataStorageSelectors from '../selectors/dataStorageSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useDataStorageArray = () => useShallowSelector(dataStorageSelectors.dataStorageArray);
export const useDataStorageForUserInProject = (userId, projectKey) => {
  const dataStores = useDataStorageArray();
  return {
    ...dataStores,
    value: dataStores.value.filter(store => store.projectKey === projectKey && store.users.includes(userId)),
  };
};
