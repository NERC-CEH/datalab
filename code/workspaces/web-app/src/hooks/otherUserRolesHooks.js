import useShallowSelector from './useShallowSelector';
import otherUserRolesSelectors from '../selectors/otherUserRolesSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useOtherUserRoles = () => useShallowSelector(otherUserRolesSelectors.otherUserRoles);
