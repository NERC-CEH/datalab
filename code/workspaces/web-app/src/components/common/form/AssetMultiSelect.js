import React, { useState } from 'react';
import { renderMultiSelectAutocompleteField } from './controls';
import { useAssetRepo, useVisibleAssets } from '../../../hooks/assetRepoHooks';
import assetLabel from './assetLabel';

function AssetMultiSelect({ input, meta = null, showAllAssets, projectKey = undefined, ...custom }) {
  const projectAssets = useVisibleAssets(projectKey);
  const allAssets = useAssetRepo();
  const assetRepo = showAllAssets ? allAssets : projectAssets;
  const [currentValue, setCurrentValue] = useState(input.value || []); // use to give current value to onBlur

  return (
    <>
      {
      renderMultiSelectAutocompleteField(
        {
          input,
          meta,
          currentValue,
          setCurrentValue,
          options: assetRepo.value.assets,
          label: 'Assets',
          placeholder: 'Filter by asset name or location',
          getOptionLabel: assetLabel,
          getOptionSelected: (option, val) => option.assetId === val.assetId,
          loading: assetRepo.fetching,
          selectedTip: 'Asset selected',
          ...custom,
        },
      )
    }
    </>
  );
}

export default AssetMultiSelect;
