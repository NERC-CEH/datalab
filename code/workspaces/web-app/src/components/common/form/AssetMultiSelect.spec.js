import React from 'react';
import { useDispatch } from 'react-redux';
import { shallow } from 'enzyme';
import { useVisibleAssets } from '../../../hooks/assetRepoHooks';
import AssetMultiSelect from './AssetMultiSelect';

jest.mock('react-redux');
jest.mock('../../../hooks/assetRepoHooks');

const asset1 = { assetId: 'asset-1', name: 'Asset 1', version: 'v1' };
const asset2 = { assetId: 'asset-2', name: 'Asset 2', version: 'v2' };

describe('AssetMultiSelect', () => {
  const shallowRender = () => {
    const props = {
      input: {
        value: [asset1, asset2],
        onChange: jest.fn().mockName('onChange'),
      },
    };

    return shallow(<AssetMultiSelect {...props} />);
  };

  beforeEach(() => {
    useDispatch.mockReturnValue(jest.fn().mockName('dispatch'));
    useVisibleAssets.mockReturnValue({ fetching: false, value: { assets: [asset1, asset2] } });
  });

  it('renders to match snapshot passing correct props to children', () => {
    expect(shallowRender()).toMatchSnapshot();
  });
});
