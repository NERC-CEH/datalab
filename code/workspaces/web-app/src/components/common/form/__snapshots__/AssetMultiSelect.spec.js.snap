// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetMultiSelect renders to match snapshot passing correct props to children 1`] = `
<div>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
    role="combobox"
    style="min-width: 250px; margin-bottom: 10px;"
  >
    <div
      class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1w68iww-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="mui-2"
        id="mui-2-label"
      >
        Assets
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-sndw66-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-183wrp3-MuiButtonBase-root-MuiChip-root"
          data-tag-index="0"
          role="button"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Asset 1:v1 (no local file)
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconOutlinedColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CancelIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
            />
          </svg>
        </div>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-183wrp3-MuiButtonBase-root-MuiChip-root"
          data-tag-index="1"
          role="button"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Asset 2:v2 (no local file)
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconOutlinedColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CancelIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
            />
          </svg>
        </div>
        <input
          aria-autocomplete="list"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-12yjm75-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-2"
          placeholder="Filter by asset name or location"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
            tabindex="-1"
            title="Clear"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1z7n62"
          >
            <span>
              Assets
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
  </div>
</div>
`;
